src/main/kotlin/no/nav/familie/ks/sak/kjerne/beregning/TilkjentYtelseValidator.kt

```diff
@@ -74,4 +89,151 @@ object TilkjentYtelseValidator {
+
+        val aktørIder =
+            hentAktørIderForDenneOgForrigeBehandling(andelerTilkjentYtelse, forrigeAndelerTilkjentYtelse)
+
+        val personerMedUgyldigEtterbetaling =
+            aktørIder.mapNotNull { aktørId ->
+                val andelerTilkjentYtelseForPerson = andelerTilkjentYtelse.filter { it.aktør.aktørId == aktørId }
+                val forrigeAndelerTilkjentYtelseForPerson =
+                    forrigeAndelerTilkjentYtelse?.filter { it.aktør.aktørId == aktørId }
+
+                val etterbetalingErUgyldig = erUgyldigEtterbetalingPåPerson(
+                    forrigeAndelerTilkjentYtelseForPerson,
+                    andelerTilkjentYtelseForPerson,
+                    gyldigEtterbetalingFom
+                )
+
+                if (etterbetalingErUgyldig) {
+                    aktørId
+                } else {
+                    null
+                }
+            }
+
+        return personerMedUgyldigEtterbetaling
+    }
+
+    fun hentAktørIderForDenneOgForrigeBehandling(
+        andelerTilkjentYtelse: List<AndelTilkjentYtelse>,
+        forrigeAndelerTilkjentYtelse: List<AndelTilkjentYtelse>?
+    ): Set<String> {
+        val aktørIderFraAndeler = andelerTilkjentYtelse.map { it.aktør.aktørId }
+        val aktøerIderFraForrigeAndeler = forrigeAndelerTilkjentYtelse?.map { it.aktør.aktørId } ?: emptyList()
+        return (aktørIderFraAndeler + aktøerIderFraForrigeAndeler).toSet()
+    }
+
+    fun erUgyldigEtterbetalingPåPerson(
+        forrigeAndelerForPerson: List<AndelTilkjentYtelse>?,
+        andelerForPerson: List<AndelTilkjentYtelse>,
+        gyldigEtterbetalingFom: YearMonth?
+    ): Boolean {
+        return YtelseType.values().any { ytelseType ->
+            val forrigeAndelerForPersonOgType = forrigeAndelerForPerson?.filter { it.type == ytelseType } ?: emptyList()
+            val andelerForPersonOgType = andelerForPerson.filter { it.type == ytelseType }
+
+            val forrigeAndelerTidslinje = forrigeAndelerForPersonOgType.map {
+                Periode(
+                    it,
+                    it.stønadFom.toLocalDate(),
+                    it.stønadTom.toLocalDate()
+                )
+            }.tilTidslinje()
+            val andelerTidslinje =
+                andelerForPersonOgType.map { Periode(it, it.stønadFom.toLocalDate(), it.stønadTom.toLocalDate()) }
+                    .tilTidslinje()
+
+            val erAndelMedØktBeløpFørGyldigEtterbetalingsdato =
+                erAndelMedØktBeløpFørDato(
+                    forrigeAndeler = forrigeAndelerForPersonOgType,
+                    andeler = andelerForPersonOgType,
+                    måned = gyldigEtterbetalingFom
+                )
+
+            val segmenterLagtTil =
+                andelerTidslinje.kombinerMed(forrigeAndelerTidslinje) { andel1, andel2 ->
+                    if (andel2 == null) andel1 else null
+                }.tilPerioder().filtrerIkkeNull()
```

Comment:
```suggestion
                }.tilPerioderIkkeNull()
```
Vi har en tilPerioderIkkeNull som gjør det samme 👍